### ISSUE TITLE
Learning rate interval update not working properly

### ISSUE THREAD

**Issue by fmellomascarenhas** [PyTorchLightning/pytorch-lightning#6948]
## 🐛 Bug

When I use Onecycle scheduler, it only works properly if I set `steps_per_epoch = 1`, even though I have set 'interval': 'step'.

```
scheduler = torch.optim.lr_scheduler.OneCycleLR(optimizer,
                                                            max_lr,
                                                            epochs,
                                                            steps_per_epoch = len(self.loaders_dict['train']))

return {"optimizer": optimizer, "lr_scheduler": scheduler, "monitor": "train_loss", 'interval': 'step'}
```

Here is the learning rate at the end of training if I set steps_per_epoch > 1
![image](https://user-images.githubusercontent.com/23004953/114280619-1eacf780-9a08-11eb-9f6c-d8b447eafcd3.png)

I went to check the code:

Here is the [training loop](https://github.com/PyTorchLightning/pytorch-lightning/blob/851f9e3997db8d8e077254e3a996b8835d59dc3c/pytorch_lightning/trainer/training_loop.py#L403)

There seems to be two main functions that update the learning rate:
[here](https://github.com/PyTorchLightning/pytorch-lightning/blob/851f9e3997db8d8e077254e3a996b8835d59dc3c/pytorch_lightning/trainer/training_loop.py#L455)
`self.update_train_loop_lr_schedulers(monitor_metrics=monitor_metrics)` 

that later calls:
`self.trainer.optimizer_connector.update_learning_rates(interval="step", monitor_metrics=monitor_metrics)`

and
[here](https://github.com/PyTorchLightning/pytorch-lightning/blob/851f9e3997db8d8e077254e3a996b8835d59dc3c/pytorch_lightning/trainer/training_loop.py#L492)
`self.trainer.optimizer_connector.update_learning_rates(interval='epoch')` 

I cant figure out what exactly is wrong, any help will be appreciated. Thanks!

 - PyTorch Version 1.7
 - Linux

**Comment by fmellomascarenhas** [comment]
Nevermind, I just realized that 'interval' needs to be inside of the scheduler

schedulers = [
      {
         'scheduler': ReduceLROnPlateau(optimizers[0], ...),
         'monitor': 'metric_to_track',
         'interval': 'step',
         'frequency': 10,
         'strict': True,
      },
      LambdaLR(optimizers[1], ...)
   ]
